### DISCUSSION TITLE
The theme does not work in the Bedrock project

### DISCUSSION THREAD

**Discussion by Sergey98Am** [gregsullivan/_tw#175] [General]
I have a WordPress Bedrock project, but this theme does not work there. In another WordPress project without Bedrock, it is working.

![image (6)](https://github.com/gregsullivan/_tw/assets/52923458/03cc54cd-327e-4762-9324-2f1ef7e6b9e2)

I get these `style.css` and `script.min.js`  404 errors.

![Screenshot from 2024-05-23 18-32-44](https://github.com/gregsullivan/_tw/assets/52923458/a698dad6-3575-4c04-9e97-53794b594625)

Is there a way to use this theme in the Bedrock project?

**Comment by gregsullivan** [comment]
Hi Sergey—Thanks for posting this!

I've never used Bedrock before, so I can't confirm one way or another. I don't remember hearing from anyone saying they were using it with _tw.

Can you confirm that you've run `npm run dev` (or `watch` or `prod`)? I just want to confirm the `script.min.js` and `style.css` files have definitely been created.

Barring that, I would need to set up Bedrock and see what I found during that process. I won't have time to do that in the immediate future, but I can keep this open and make an attempt when possible!

**Comment by Sergey98Am** [comment]
@gregsullivan. Yes, I have run these commands, there are `script.min.js` and `style.css` files

![Screenshot from 2024-05-24 14-30-01](https://github.com/gregsullivan/_tw/assets/52923458/582ac85a-8663-4ff7-9ce2-936e8bbad874)

**Comment by gregsullivan** [comment]
Great, thank you! I'll try to get Bedrock set up locally, and I'll note anything I find here.

**Comment by Sergey98Am** [comment]
Thanks in advance

**Comment by kyle-cyberprairie** [comment]
@Sergey98Am is this because of the file structure? I am using _tw with Bedrock and it is working for me. I had to set a symlink for my theme, so ```/my-theme``` exists on the project root, which ```web/app/themes/my-theme``` points to.

**Comment by gregsullivan** [comment]
I'm revisiting this tonight, and I believe agreeing with this answer is as confident as I can be in terms of the best approach:

> @Sergey98Am is this because of the file structure? I am using _tw with Bedrock and it is working for me. I had to set a symlink for my theme, so /my-theme exists on the project root, which web/app/themes/my-theme points to.

A couple of points:

- My understanding is that Bedrock users are likely to be deploying the same structure from development to production. With that being the case, keeping the theme repository from _tw in the project root and symlinking `./my-theme/theme/` to `./web/app/themes/my-theme` would be preferable in that you're only keeping the deployable portion of _tw in the public folder.
- I continue not to have personal experience with Bedrock, so I can't state definitively that symlinking within a Bedrock project isn't discouraged for other reasons. But I don't foresee any issues with that approach, and I still use symlinking without issue on local development environments for some if the _tw-based sites that I build.

I'm going to convert this to a discussion, and I'd be happy to revisit this further if anyone has questions about or issues with this in the future.
